lib/url.c

```diff
@@ -677,28 +677,30 @@ static bool conn_maxage(struct Curl_easy *data,
                         struct connectdata *conn,
                         struct curltime now)
 {
-  timediff_t idletime, lifetime;
+  timediff_t age_ms;
 
-  idletime = curlx_timediff(now, conn->lastused);
-  idletime /= 1000; /* integer seconds is fine */
+  if(data->set.maxage_conn) {
+    age_ms = curlx_timediff(now, conn->lastused);
 
-  if(idletime > data->set.maxage_conn) {
-    infof(data, "Too old connection (%" FMT_TIMEDIFF_T
-          " seconds idle), disconnect it", idletime);
-    return TRUE;
+    if(age_ms > (data->set.maxage_conn * 1000)) {
```

Comment:
If `maxage_conn` was set to a ridiculous value (> LONG_MAX/1000), this might integer overflow and misbehave.